cmd/earthly/main.go

```diff
@@ -1176,6 +1176,86 @@ Set up a whole custom git repository for a server called example.com, using a si
 				},
 			},
 		},
+		{
+			Name: "preview",
+			Subcommands: []*cli.Command{
+				{
+					Name:        "project",
+					Description: "Create and manage Earthly projects",
+					UsageText:   "earthly project (ls|rm|create|member)",
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "org",
+							EnvVars:  []string{"EARTHLY_ORG"},
+							Usage:    "The name of the organization to which the project belongs. Required when user is a member of multiple.",
+							Required: false,
+						},
+					},
+					Subcommands: []*cli.Command{
+						{
+							Name:        "ls",
+							Aliases:     []string{"list"},
+							Usage:       "List all projects that belong to the specified organization",
+							Description: "List all projects that belong to the specified organization",
+							UsageText:   "earthly project --org <organization-name> ls",
+							Action:      app.actionProjectList,
+						},
+						{
+							Name:        "rm",
+							Aliases:     []string{"remove", "delete"},
+							Usage:       "Remove an existing project from the organization",
+							Description: "Remove an existing project from the organization",
+							UsageText:   "earthly project --org <organization-name> rm <project-name>",
+							Action:      app.actionProjectRemove,
+						},
+						{
+							Name:        "create",
+							Usage:       "Create a new project in the specified organization",
+							Description: "Create a new project in the specified organization",
+							UsageText:   "earthly project --org <organization-name> create <project-name>",
+							Action:      app.actionProjectCreate,
+						},
+						{
+							Name:        "member",
+							Description: "Manage project members",
+							UsageText:   "earthly project member (ls|rm|add|update)",
+							Subcommands: []*cli.Command{
+								{
+									Name:        "add",
+									Usage:       "Add a new member to the specified project",
+									Description: "Add a new member to the specified project",
+									UsageText:   "earthly project --org <organization-name> member add <project-name> <user-id-or-email> <permission>",
+									Action:      app.actionProjectMemberAdd,
+								},
+								{
+									Name:        "rm",
+									Aliases:     []string{"remove", "delete"},
+									Usage:       "Remove a member from the specified project",
+									Description: "Remove a member from the specified project",
+									UsageText:   "earthly project --org <organization-name> member rm <project-name> <user-id>",
```

Comment:
it still feels a bit odd that we can't use an `email` here. Can we create a ticket/issue/task to implement it down the road?

Reply:
We can normalize these commands and the endpoints to use email if that's favorable. I feel a little apprehensive about using email addresses in place of user IDs or names as emails are usually treated as semi-private (PII, I believe), but we're already using them elsewhere so I'm not sure if that's a concern.

Reply:
ok, lets punt on it. Maybe we'll need a notion of usernames if emails-are-PII becomes an issue.